qemu/tests/block_resize.py

```diff
@@ -1,171 +1,148 @@
-    accept_ratio = float(params.get("accept_ratio", 0))
-
-    error_context.context("Check image size in guest", logging.info)
-    block_size = get_block_size(session, block_size_cmd, block_size_pattern)
-    if (block_size > data_image_size or
-            block_size < data_image_size * (1 - accept_ratio)):
-        test.error("Image size from guest and image not match"
-                   "Block size get from guest: %s \n"
-                   "Image size get from image: %s \n"
-                   % (block_size, data_image_size))
-
-    if params.get("guest_prepare_cmd"):
-        session.cmd(params.get("guest_prepare_cmd"))
-
-    if params.get("format_disk", "no") == "yes":
-        error_context.context("Format disk", logging.info)
-        utils_misc.format_windows_disk(session, params["disk_index"],
-                                       mountpoint=params["disk_letter"])
-
-    disk_update_cmd = params.get("disk_update_cmd")
-    if disk_update_cmd:
-        disk_update_cmd = disk_update_cmd.split("::")
-
-    disk_rescan_cmd = params.get("disk_rescan_cmd")
-
-    block_size = data_image_size
-    disk_change_ratio = params["disk_change_ratio"]
-    for index, ratio in enumerate(disk_change_ratio.strip().split()):
-        old_block_size = block_size
-        block_size = int(int(data_image_size) * float(ratio))
-
-        if block_size == old_block_size:
-            logging.warn("Block size is not changed in round %s."
-                         " Just skip it" % index)
-            continue
-
-        if disk_update_cmd:
-            if "DISK_CHANGE_SIZE" in disk_update_cmd[index]:
-                disk_unit = params.get("disk_unit", "M")
-                size = abs(block_size - old_block_size)
-                change_size = utils_misc.normalize_data_size("%sB" % size,
-                                                             disk_unit)
-                disk_update_cmd[index] = re.sub("DISK_CHANGE_SIZE",
-                                                change_size.split(".")[0],
-                                                disk_update_cmd[index])
-
-        # So far only virtio drivers support online auto block size change in
-        # linux guest. So we need manully update the the disk or even reboot
-        # guest to get the right block size after change it from monitor.
+            disk = utils_disk.get_windows_disks_index(session, img_size)[0]
+            utils_disk.update_windows_disk_attributes(session, disk)
+        error_context.context("Formatting disk", logging.info)
+        mpoint = utils_disk.configure_empty_disk(session, disk, img_size,
+                                                 os_type, fstype=fstype,
+                                                 labeltype=labeltype)[0]
+        partition = mpoint.replace('mnt', 'dev') if 'mnt' in mpoint else None
+
+    for ratio in params.objects("disk_change_ratio"):
+        block_size = int(int(block_virtual_size) * float(ratio))
```

Comment:
Why force to convert the type of `int(block_virtual_size) * float(ratio)` to `int`?

Reply:
uh, firstly, it's from original line 119, and secondly, at the end of our test, we need to judge block_size if match size return from utils_disk. get_disk_size(), both int type for this func return, I changed the get_disk_size() return type  from float to int because the xuhan's comment before, you can check it again~ thanks